### PR TITLE
Create mungegithub 'Server' feature

### PR DESCRIPTION
**Pull request by cjwagner**
This lets mungers share an http server and mux instead of only the submit-queue or cherry-pick munger running a server.  This will also allow any munger to expose prometheus metrics easily since a server will already be running.

Note: The first commit in this PR that deletes the bulk-lgtm munger has its own PR (#4101).

/cc @spiffxp @rmmh

### CODE REVIEW

File: mungegithub/mungers/submit-queue.go [hunk 1]
```diff
@@ -474,40 +467,34 @@ func (sq *SubmitQueue) internalInitialize(config *github.Config, features *featu
...
-		http.Handle("/merge-info", gziphandler.GzipHandler(http.HandlerFunc(sq.serveMergeInfo)))
-		http.Handle("/priority-info", gziphandler.GzipHandler(http.HandlerFunc(sq.servePriorityInfo)))
-		http.Handle("/health", gziphandler.GzipHandler(http.HandlerFunc(sq.serveHealth)))
-		http.Handle("/health.svg", gziphandler.GzipHandler(http.HandlerFunc(sq.serveHealthSVG)))
-		http.Handle("/sq-stats", gziphandler.GzipHandler(http.HandlerFunc(sq.serveSQStats)))
-		http.Handle("/flakes", gziphandler.GzipHandler(http.HandlerFunc(sq.serveFlakes)))
-		http.Handle("/metadata", gziphandler.GzipHandler(http.HandlerFunc(sq.serveMetadata)))
+	if features.Server.Enabled {
+		features.Server.Handle("/prs", gziphandler.GzipHandler(http.HandlerFunc(sq.servePRs)))
+		features.Server.Handle("/history", gziphandler.GzipHandler(http.HandlerFunc(sq.serveHistory)))
```
Comment (BenTheElder) [inline mungegithub/mungers/submit-queue.go]: Something I've wondered before, aren't these behind nginx which can do gzip?

**Comment by rmmh** [comment]
/lgtm

This seems like a pretty straightforward refactor.

File: mungegithub/mungers/e2e/e2e.go [hunk 2]
```diff
@@ -75,9 +75,9 @@ func (e *RealE2ETester) Init(adminMux HTTPHandlerInstaller) *RealE2ETester {
 	e.flakeCache = cache.NewCache(e.getGCSResult)
 	e.resolutionTracker = NewResolutionTracker()
 	if adminMux != nil {
-		adminMux.HandleFunc("/api/mark-resolved", e.resolutionTracker.SetHTTP)
-		adminMux.HandleFunc("/api/is-resolved", e.resolutionTracker.GetHTTP)
-		adminMux.HandleFunc("/api/list-resolutions", e.resolutionTracker.ListHTTP)
+		adminMux.Handle("/api/mark-resolved", http.HandlerFunc(e.resolutionTracker.SetHTTP))
```
Comment (rmmh) [inline mungegithub/mungers/e2e/e2e.go]: Can we implement HandleFunc? It looks like nice sugar.

File: mungegithub/mungers/submit-queue.go (hunk 1 above)
Comment (rmmh) [inline mungegithub/mungers/submit-queue.go]: That would be cleaner. We could turn this into a bunch of HandleFunc calls and put the GzipHandler at the proxy layer.

Other deployments might have a different nginx setup, though.

File: mungegithub/mungers/e2e/e2e.go (hunk 2 above)
Comment (cjwagner) [inline mungegithub/mungers/e2e/e2e.go]: Done.

**Comment by cjwagner** [comment]
I'll wait for #4101 to merge then I'll rebase and merge this PR.

**Pull request merged**